internal/extproc/backendauth/azure.go

```diff
@@ -0,0 +1,102 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package backendauth
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	corev3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	extprocv3 "github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3"
+
+	"github.com/envoyproxy/ai-gateway/filterapi"
+)
+
+type azureHandler struct {
+	azureAccessToken string
+}
+
+func newAzureHandler(azureAuth *filterapi.AzureAuth) (Handler, error) {
+	if azureAuth != nil {
+		content, err := os.ReadFile(azureAuth.Filename)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read azure access token file: %w", err)
+		}
+		// Extract access token from secret which content a key-value string
+		// such as `azure_access_token: secret_value`.
+		lines := strings.Split(string(content), "\n")
+		var azureAccessToken string
+		for _, line := range lines {
+			parts := strings.SplitN(line, "=", 2)
+			if len(parts) == 2 && strings.TrimSpace(parts[0]) == "azure_access_token" {
+				azureAccessToken = strings.TrimSpace(parts[1])
+				break
+			}
+		}
+		if azureAccessToken == "" {
+			return nil, fmt.Errorf("azure_access_token not found in the secret file")
+		}
+		return &azureHandler{azureAccessToken: azureAccessToken}, nil
+	}
+	return nil, fmt.Errorf("azure auth configuration is required")
+}
+
+func (a *azureHandler) Do(_ context.Context, requestHeaders map[string]string, headerMut *extprocv3.HeaderMutation, bodyMut *extprocv3.BodyMutation) error {
+	requestHeaders["Authorization"] = fmt.Sprintf("Bearer %s", a.azureAccessToken)
+	headerMut.SetHeaders = append(headerMut.SetHeaders, &corev3.HeaderValueOption{
+		Header: &corev3.HeaderValue{Key: "Authorization", RawValue: []byte(requestHeaders["Authorization"])},
+	})
+
+	// lookup model name from request body
+	model, err := extractModel(bodyMut.GetBody())
```

Comment:
The body is already decoded before calling auth handler, we might want to move the path mutation outside.